### PR TITLE
[botan] Fixes for building with clang/libc++/winpthreads.

### PR DESCRIPTION
**Pull request by solemnwarning**
This change allows building the botan port using the Clang toolchain in VS2022 and a custom libc++/winpthreads runtime (custom triplet based on https://github.com/Neumann-A/my-vcpkg-triplets/blob/master/x86-win-llvm.cmake), I have verified the port still builds using the vanilla MSVC compiler and runtime.

I'm not a regular contributor to vcpkg, so please let me know if any of the things I've changed here should be done differently.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by solemnwarning** [comment]
It appears I did something wrong when updating the port versions here and in #36954, but not sure what - I ran the  `vcpkg x-add-version --all` and committed the result along with each of my changes. Can anyone advise?

Thanks

File: ports/botan/libcxx-winpthread-fixes.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
+--- a/src/tests/tests.h
++++ b/src/tests/tests.h
+@@ -17,6 +17,7 @@
+ #include <map>
+ #include <memory>
+ #include <optional>
++#include <ranges>
```
Comment (dg0yt) [inline ports/botan/libcxx-winpthread-fixes.patch:7]: This line breaks android.

Comment (dg0yt) [reply ports/botan/libcxx-winpthread-fixes.patch:7]: It probably depends on NDK. vcpkg CI is still stuck at r25.

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Seems reasonable, thanks :)

**Pull request merged**